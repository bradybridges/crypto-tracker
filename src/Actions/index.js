export const updateCryptos = (cryptos) => ({
  type: 'UPDATE_CRYPTOS',
  cryptos,
});

export const updateError = (error) => ({
  type: 'UPDATE_ERROR',
  error,
});

export const updateTimeFrame = (timeFrame) => ({
  type: 'UPDATE_TIME_FRAME',
  timeFrame,
});

export const updateTrackedCoins = (symbols) => ({
  type: 'UPDATE_TRACKED_COINS',
  symbols,
});

export const updatePortfolio = (portfolio) => ({
  type: 'UPDATE_PORTFOLIO',
  portfolio,
});
