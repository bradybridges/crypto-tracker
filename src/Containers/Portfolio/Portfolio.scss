#portfolio-container {
  height: 70vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  background-image: url('../../Images/30\ Black\ abstract\ Wallpapers\ HD\ Download.jpg');
  color: white;
  overflow: scroll;
}

#view-value-btn {
  width: 80%;
  height: 50px;
  border-radius: 4px;
  background-color: black;
  color: white;
  font-size: 1em;
  font-weight: 800;
  &:hover {
    cursor: pointer;
  }
}