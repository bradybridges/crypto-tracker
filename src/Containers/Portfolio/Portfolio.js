import React, { Component } from 'react';
import { connect } from 'react-redux';
import { updatePortfolio } from '../../Actions/index';
import QuantityInput from '../../Components/QuantityInput/QuantityInput';
import './Portfolio.scss';

class Portfolio extends Component {
  renderInputs = () => {
    const coinInputs = this.props.cryptos.map( coin => {
      const existingCoin = this.props.portfolio.find( crypto => crypto.name === coin.name);
      const qty = existingCoin ? existingCoin.qty : 0;
      return (
        <QuantityInput key={coin.id} coin={coin} qty={qty} />
      );
    });
    return coinInputs;
  }

  render() {
    return (
      <section id='portfolio-container'>
        {this.renderInputs()}
        <button>View Value</button>
      </section>
    );
  }
}

const mapStateToProps = state => ({
  cryptos: state.cryptos,
  portfolio: state.portfolio,
});

const mapDispatchToProps = dispatch => ({
  updatePortfolio: portfolio => dispatch( updatePortfolio(portfolio) ),
});

export default connect(mapStateToProps, mapDispatchToProps)(Portfolio);