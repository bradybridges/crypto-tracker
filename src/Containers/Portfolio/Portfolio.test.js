import React from 'react';
import { shallow } from 'enzyme';
import { Portfolio, mapDispatchToProps, mapStateToProps } from './Portfolio';
import { updatePortfolio } from '../../Actions/index';

describe('Portfolio', () => {
  let wrapper;
  const mockCryptos = [
    {
      currecy: 'BTC',
      id: 'BTC',
      price: 9341.6987,
      price_date: '2019-11-02',
      symbol: 'BTC',
      circulating_supply: '18025852',
      max_supply: '21000000',
      name: 'Bitcoin',
      logo_url: 'https://s3.us-east-2.amazonaws.com/nomics-api/static/images/currencies/btc.svg',
      market_cap: '1243423423.05',
      rank: '1',
      high: '19480.90',
      high_timestamp: '2017-12-16',
    },
    {
      currecy: 'LTC',
      id: 'LTC',
      price: 75.55,
      price_date: '2019-11-02',
      symbol: 'LTC',
      circulating_supply: '18000000',
      max_supply: '20000000',
      name: 'Litecoin',
      logo_url: 'https://s3.us-east-2.amazonaws.com/nomics-api/static/images/currencies/ltc.svg',
      market_cap: '1003423423.05',
      rank: '3',
      high: '420.90',
      high_timestamp: '2017-12-10',
    },
  ];

  const mockPortfolio = [{name: 'Bitcoin', qty: 10}, {name: 'Litecoin', qty: 3}];
  beforeEach(() => {
    wrapper = shallow(<Portfolio cryptos={mockCryptos} portfolio={mockPortfolio} />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('mapStateToProps should return cryptos and portfolio', () => {
    const mockState = {
      cryptos: mockCryptos,
      portfolio: mockPortfolio,
      error: 'There was an error',
    };
    const expected = {
      cryptos: mockCryptos,
      portfolio: mockPortfolio,
    }
    const result = mapStateToProps(mockState);
    expect(result).toEqual(expected);
  });

  it('updatePortfolio should dispatch updatePortfolio(portfolio)', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = updatePortfolio(mockPortfolio);
    const mappedDispatch = mapDispatchToProps(mockDispatch);
    mappedDispatch.updatePortfolio(mockPortfolio);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
