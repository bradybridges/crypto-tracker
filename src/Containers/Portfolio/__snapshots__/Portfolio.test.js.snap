// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio should match the snapshot 1`] = `
<section
  id="portfolio-container"
>
  <h2>
    Portfolio
  </h2>
  <Connect(QuantityInput)
    coin={
      Object {
        "circulating_supply": "18025852",
        "currecy": "BTC",
        "high": "19480.90",
        "high_timestamp": "2017-12-16",
        "id": "BTC",
        "logo_url": "https://s3.us-east-2.amazonaws.com/nomics-api/static/images/currencies/btc.svg",
        "market_cap": "1243423423.05",
        "max_supply": "21000000",
        "name": "Bitcoin",
        "price": 9341.6987,
        "price_date": "2019-11-02",
        "rank": "1",
        "symbol": "BTC",
      }
    }
    key="BTC"
    qty={10}
  />
  <Connect(QuantityInput)
    coin={
      Object {
        "circulating_supply": "18000000",
        "currecy": "LTC",
        "high": "420.90",
        "high_timestamp": "2017-12-10",
        "id": "LTC",
        "logo_url": "https://s3.us-east-2.amazonaws.com/nomics-api/static/images/currencies/ltc.svg",
        "market_cap": "1003423423.05",
        "max_supply": "20000000",
        "name": "Litecoin",
        "price": 75.55,
        "price_date": "2019-11-02",
        "rank": "3",
        "symbol": "LTC",
      }
    }
    key="LTC"
    qty={3}
  />
  <button
    id="view-value-btn"
    onClick={[Function]}
  >
    View Value
  </button>
</section>
`;
